subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/AbstractPropertyNode.java

```diff
@@ -18,7 +18,7 @@
 
 import javax.annotation.Nullable;
 
-abstract class AbstractPropertyNode<SELF extends AbstractPropertyNode<SELF>> implements PropertyNode<SELF> {
+public abstract class AbstractPropertyNode implements PropertyNode {
```

Comment:
We weren't even using `SELF`? :o

Reply:
Now we don't use it anymore, since we are not returning an `Iterator<SELF>`.

Reply:
:+1: I'm leaning more and more towards considering `<SELF>` a code smell.